### DISCUSSION TITLE
BTT Filament Sensor Ender 3 pro v2

### DISCUSSION THREAD

**Discussion by liz-muehlmann** [mriscoc/Ender3V2S1#1000] [Q&A]
Board: Creality 4.2.2
Professional Firmware: v. 2.1.3, 20221222 - 2:12
BTT Smart Filament Sensor

I have the filament sensor wired directly to the mainboard. I have enabled runout from the Advanced screen and set RUNOUT ACTIVE to Motion. I have also tried it on High based on another person's comments. Neither have solved the problem. 

Using the end stop diagnostics, I started manually pulling filament through. The filament option will toggle between TRIGGERED and OPEN when there is filament running through the sensor. 

When a print is running and the Runout Active option is set to Motion, it will immediately park the nozzle for a filament change. When it's set to High, the print gets a little further, but it still triggers when there is filament running through. +

Based on this [discussion](https://github.com/mriscoc/Ender3V2S1/discussions/907), I've tried both a bed level print and a larger one and both have the same problem.

My slicer settings follow what is listed in the wiki (the settings below are copy and pasted from Cura).

This is a brand new BTT sensor and the second one I have tried. Can anyone help me troubleshoot? 

Start G-Code
`M201 X500.00 Y500.00 Z100.00 E5000.00 ;Setup machine max acceleration
M203 X500.00 Y500.00 Z20.00 E50.00 ;Setup machine max feedrate
M204 P500.00 R1000.00 T500.00 ;Setup Print/Retract/Travel acceleration
M205 X8.00 Y8.00 Z0.40 E5.00 ;Setup Jerk
M220 S100 ;Reset Feedrate
M221 S100 ;Reset Flowrate

G92 E0 ; Reset Extruder
G28 ; Home all axes
M420 S1 Z2; Use a mesh leveling up to 2mm (a valid mesh must be loaded into RAM);
G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X0.1 Y20 Z0.3 F5000.0 ; Move to start position
G1 X0.1 Y200.0 Z0.3 F1500.0 E15 ; Draw the first line
G1 X0.4 Y200.0 Z0.3 F5000.0 ; Move to side a little
G1 X0.4 Y20 Z0.3 F1500.0 E30 ; Draw the second line
G92 E0 ; Reset Extruder
G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X5 Y20 Z0.3 F5000.0 ; Move over to prevent blob squish`

End G-Code:
`G91 ;Relative positioning
G1 E-2 F2700 ;Retract a bit
G1 E-2 Z0.2 F2400 ;Retract and raise Z
G1 X5 Y5 F3000 ;Wipe out
G1 Z10 ;Raise Z more
G90 ;Absolute positioning

G1 X0 Y200 ;Present print
M106 S0 ;Turn-off fan
M104 S0 ;Turn-off hotend
M140 S0 ;Turn-off bed
M84 ;Disable steppers`

**Comment by Rippeee** [comment]
Have you found any solution yet? I'm having the same issue.

**Comment by Belliqueu** [comment]
I just upgraded to the latest firmware and started having this same problem on my Ender 3 S1

**Comment by Rippeee** [comment]
For those who are having issues with BTT sensor triggering every 30sec, I think I found the solution. I'm not using any custom configs or G-godes. I'm using the latest mriscoc firmware with BTT sensor, Mode set as motion. The recommended Distance for BTT is 7 mm, and that doesn't work, unless your sensor is literally Next to the extruder. I would recommend to set it to 14 mm, if that doesn't work, try 20,30,40,50 etc. In my case, I'm running Sprite pro Direct drive upgrade kit, and I'm mounting the sensor at the top, next to the filament spool. For that reason, there is a lot of ''slag'' between the extruder and the sensor, so there will be moments when the filament ain't moving at all. 7 mm would never work in my case and in my case it doesn't really matter even if the distance is 100 mm because once the filament is out of sensor, there is still like 20CM of filament going in the extruder. In a Nutshell the solution is Crank up your distance from 7 to 8,9,10,11,12,13,14,15,16… etc.   Until it works.

**Comment by Rippeee** [comment]
I made a mounting model to fix this issue, with this model you can keep the original 7 mm Run out distance.

You are welcome…  https://www.thingiverse.com/thing:6209226
